File: C/0013_roman_to_integer/0013_roman_to_integer.c
int romanToInt(char * s) {
    int hashmap[256];
    memset(hashmap, 0, sizeof(int) * 256);
    hashmap[(int)('I')] = 1;
    hashmap[(int)('V')] = 5;
    hashmap[(int)('X')] = 10;
    hashmap[(int)('L')] = 50;
    hashmap[(int)('C')] = 100;
    hashmap[(int)('D')] = 500;
    hashmap[(int)('M')] = 1000;
    
    int result = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (i == 0) {
            result += hashmap[s[i]];
        }
        if (i > 0) {
            if (hashmap[s[i]] <= hashmap[s[i - 1]]) {
                result += hashmap[s[i]];
            }
            else {
                result = result - 2 * hashmap[s[i - 1]];
                result += hashmap[s[i]];
            }
        }
    }
    return result;
}

File: Java/0013_roman_to_integer/0013_roman_to_integer.java
package _13_Roman_to_Integer;

import java.util.HashMap;
import java.util.Map;

public class Roman {
    public int romanToInt(String s) {
        int result = 0;
        char[] str = s.toCharArray();
        if (s.length() <= 0) {
           return result;
        }
        
        Map<Character, Integer> number = new HashMap<Character, Integer>();
        number.put('I', 1);
        number.put('V', 5);
        number.put('X', 10);
        number.put('L', 50);
        number.put('C', 100);
        number.put('D', 500);
        number.put('M', 1000);
        
        for (int i = 0; i < s.length(); i++) {
            if(i == 0)
                result += number.get(str[i]);
            if (i > 0) {
                if (number.get(str[i]) <= number.get(str[i - 1])) {
                    result += number.get(str[i]);
                }
                else {
                    result = result - 2 * number.get(str[i - 1]);
                    result += number.get(str[i]);
                }
            }
        }   
        return result;
    }
}
